src/main/io/osd.c

```diff
@@ -3139,13 +3139,14 @@ static void osdShowStatsPage2(void)
 
     if (feature(FEATURE_CURRENT_METER)) {
         displayWrite(osdDisplayPort, statNameX, top, "MAX CURRENT      :");
-        itoa(stats.max_current, buff, 10);
+        osdFormatCentiNumber(buff, stats.max_current, 0, 2, 0, 3);
         tfp_sprintf(buff, "%s%c", buff, SYM_AMP);
         displayWrite(osdDisplayPort, statValuesX, top++, buff);
 
         displayWrite(osdDisplayPort, statNameX, top, "MAX POWER        :");
-        itoa(stats.max_power, buff, 10);
-        tfp_sprintf(buff, "%s%c", buff, SYM_WATT);
+        bool kiloWatt = osdFormatCentiNumber(buff, stats.max_power, 1000, 2, 2, 3);
+        buff[3] = kiloWatt ? SYM_KILOWATT : SYM_WATT;
+        buff[4] = '\0';
```

Comment:
For some reason, the scaling to kW doesn't work here, while it does work in the regular OSD element. Anyone know why?

Reply:
IDK it is weird, I can't see the issue. Did you find why ?

Reply:
No, not yet. `osdFormatCentiNumber` does correctly format the max power level but the symbol doesn't change. I'll experiment with it some more later today.

Reply:
I don't understand why this doesn't work. `osdFormatCentiNumber` correctly formats the value when power <1000W but not when it goes above that, and apparently it does not return `true` either. It has to be withing will the `scale` then but I don't see why that only causes problems here and not in the osd element.

Reply:
@DzikuVx Do you have an idea why this doesn't work?

Reply:
Pushed the fix, it was an overflow in the stats recording